[WIP] [RFC]  RA-family Renesas EK-RA4M1 and EK-RA6M5 boards preliminary support.

piotrrak opened the pull request:
Hello
There is still many thing, but since other board renesas  ra support was merged, I though  to share to get a feedback sooner than later.

Still lot to be done, not full list of major items:
- [ ] Boards documentation
- [ ] Discuss how to handle odd-ball flash memory in DTS (similar to amtel,sam in some respects)
- [ ] Clocks (CGC driver needs extending for RA6/RA8 series support)
- [ ] pinctrl pins names for functions we probably generate those by some kind of script, writing those by hand is pretty tedious and err-prone - RA8M1/RA8M1 have 224 pins with lot more functions.

nordicjm committed:
soc: snps_qemu: Port to HWMv2

Ports the snps_qemu SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: qemu_arc: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_nsim: Port to HWMv2

Ports the snps_nsim SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: nsim: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_hsdk: Port to HWMv2

Ports the snps_arc_hsdk SoC configuration to hardware model
version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: hsdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_emsk: Port to HWMv2

Ports the snps_emsk SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: em_starterkit: Convert to v2

Converts the board to hwmv2, documentation for this board needs to
be vastly updated

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Update board names for hwmv2

Updates tests that use board names which have changed with boards v2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_emsdp: Port to HWMv2

Ports the snps_emsdp SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: emsdp: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_iot: Port to HWMv2

Ports the snps_arc_iot SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: iotdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_hsdk4xd: Port to HWMv2

Ports the snps_arc_hsdk4xd SoC configuration to hardware model
version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: hsdk4xd: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: psoc6 and psoc_6: Port to HWMv2

Ports the psoc6 and psoc_6 SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8ckit_062_ble: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8ckit_062_wifi_bt: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

golowanow committed:
soc: xtensa: intel_adsp: move and convert to HWMv2

Move and convert soc/xtensa/intel_adsp SoC family configurations
to HWMv2 with its SoC series:
`ace` (INTEL_ACE) and `cavs` (INTEL_ADSP_CAVS).

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
boards: arm: cy8ckit_062s4: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8cproto_062_4343w: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8cproto_063_ble: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

golowanow committed:
boards: xtensa: intel_adsp_cavs25: move and convert to HWMv2

Move and convert to HWMv2 intel_adsp_cavs25 board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: xtensa: intel_adsp_cavs25_tgph: change to board variant

Change `intel_adsp_cavs25_tgph` board definition to be HWMv2 board
variant `intel_adsp_cavs25/intel_tgl_adsp/tgph`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: xtensa: intel_adsp_ace15_mtpm: move and convert to HWMv2

Move and convert to HWMv2 intel_adsp_ace15_mptm board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: xtensa: intel_adsp_ace20_lnl: move and convert to HWMv2

Move and convert to HWMv2 intel_adsp_ace20_lnl board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
soc: xmc_4xxx: Port to HWMv2

Ports the xmc_4xxx SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: xmc45_relax_kit: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: xmc47_relax_kit: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
manifest: Update hal_infineon

Updates hal_infineon with changes needed for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

golowanow committed:
soc: xtensa: intel_adsp: HWMv2 workaround for SOF config

Apply a workaround at Kconfig.defconfig for CORE_COUNT default
value set overriding SOF arch/host incorrect defaults chosen
without ARCH specified by the changed HWMv2 cmake sequence.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
soc: renesas: rzt2m: Move folder structure for more SoCs

Updates the folder structure to allow for more SoCs to be added,
and fixes some minor issues

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: rcar: Port to HWMv2

Ports the rcar SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: smartbond: Port to HWMv2

Ports the smartbond SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: ra: Port to HWMv2

Ports the ra SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: arduino_uno_r4: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: da14695_dk_usb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: da1469x_dk_pro: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: rcar_spider_cr52: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
manifest: Update hal_renesas

Updates for changes with hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: rcar_h3_salvatorx_cr7: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand committed:
cmake: improved board handling for revisions

This commit improves board handling for boards in HWMv2.
On a CMake rerun, then BOARD_DIR is passed to `list_boards.py` which
is extended to take such parameter.

This allows to run `list_boards.py` whenever CMake reruns without the
penalty of searching for all board.yml files, as only the board.yml of
the current BOARD_DIR is processed.

This allows `list_boards.py` to be invoked and from there obtain list
of valid revisions and board identifiers for further board validation.

This removes the need for caching additional CMake variables related to
the board identifier and revision and thereby remove the risk of
settings becoming out of sync as only the board provided by user is
needed.

This work further ensure that use-cases described in #50536 is still
supported.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

nordicjm committed:
boards: arm64: rcar_salvator_xs_m3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand committed:
cmake: improve arch error message for invalid arch selection

Legacy hw model picked the arch based on folder names.
The new hw model allows for greater flexibility and instead uses the
architecture defined by `CONFIG_ARCH` in Kconfig.

Therefore check that ARCH is defined and fail with a better error
message if ARCH is not defined. The error message includes the board and
SoC selected to make it easier to trace such errors.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

nordicjm committed:
boards: arm and arm64: rcar_h3ulcb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

FRASTM committed:
soc: v2: stm32: Migrate STM32F2 series

Port STM32F2 series to HW model V2

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: Convert nucleo_f207zg to HWM v2

Port the board to HWMv2.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

danieldegrasse committed:
soc: nxp: kinetis: convert kinetis SOC family to hardware model V2

Convert kinetis SOC family to hardware model V2. Rework SOC Kconfig and
defconfig definitions to align with hardware model V2 guidelines.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>
Co-authored-by: Emilio Benavente <emilio.benavente@nxp.com>
Co-authored-by: David Leach <david.leach@nxp.com>
Co-authored-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
Co-authored-by: Declan Snyder <declan.snyder@nxp.com>
Co-authored-by: Yves Vandervennet <yves.vandervennet@nxp.com>

Commit:
boards: nxp: port frdm_k22f to hwmv2

Port frdm_k22f to hardware model v2

Signed-off-by: David Leach <david.leach@nxp.com>

danieldegrasse committed:
boards: nxp: port frdm_k64f to hwmv2

Port frdm_k642 to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

mmahadevan108 committed:
boards: nxp: frdm_k82f: port to hwmv2

Port frdm_k82f to hardware model v2

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Commit:
boards: nxp: convert frdm_kl25z to hwmv2

Convert frdm_kl25z to hardware model v2

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

danieldegrasse committed:
boards: nxp: frdm_kw41z: convert to hwmv2

Convert frdm_kw41z to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: convert hexiwear to hwmv2

Convert hexiwear to hardware model v2. As multiple SOCs exist on this
board, the hexiwear_k64 and hexiwear_kw40z boards are combined into one
target.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: convert rddrone_fmuk66 to hwmv2

Convert rddrone_fmuk66 to to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

Commit:
boards: nxp: twr_ke18f: convert to hwmv2

Convert twr_ke18f to hardware model v2

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
boards: nxp: convert twr_kv58f220m to hwmv2

Convert twr_kv58f220m to hardware model v2

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

danieldegrasse committed:
boards: nxp: convert usb_kw24d512 to hwmv2

Convert usb_kw24d512 to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: convert ip_k66f to hwmv2

Convert ip_k66f to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

nordicjm committed:
tests/samples/snippets: Update board names for hwmv2

Updates board names which have changed with hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

gautierg-st committed:
boards: Convert stm32l4r9i_disco to HWM v2

Port the board to HWMv2.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
boards: Convert swan_r5 to HWM v2

Port the board to HWMv2.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

fkokosinski committed:
soc/sifive/sifive_freedom: add SiFive Freedom E310 SoC

This commit adds support for the SiFive Freedom E310 SoC for the Zephyr
Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert SiFive HiFive1 to Zephyr HWMv2

This commit converts the SiFive HiFive1 board (`hifive1` target) to the
Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

gautierg-st committed:
soc: st: stm32: Migrate STM32WBA series

Port STM32WBA series to HW model v2

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
boards: Convert nucleo_wba52cg to HWM v2

Port the board to HWMv2.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
boards: Convert nucleo_wba55cg to HWM v2

Port the board to HWMv2.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

nandojve committed:
soc: atmel: Port SAM family to HWMv2

Port all the Atmel SAM SoCs to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam4e_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_due: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam4l_ek: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam4s_xplained: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam_e70_xplained: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam_v71_xult: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: RoboKit1: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samr21_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
soc: atmel: sam0: Port to HWMv2

Port all Atmel SAM0 SoCs to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samd21_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samd20_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: saml21_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samc21n_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samr34_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: same54_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

golowanow committed:
soc: riscv: intel_niosv: move and convert to HWMv2

Move to HWMv2 and convert from soc/riscv/intel_niosv SoC family
`INTEL_NIOSV` series `NIOSV` SoCs `NIOSV_M` and `NIOSV_G`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nandojve committed:
CODEOWNERS: Update Atmel entries

Update Atmel related rules to keep consistent.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
MAINTAINERS: Update Atmel entries

Update Atmel related rules to keep consistent.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: seeeduino_xiao: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: serpente: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_feather_m0_basic_proto: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_feather_m0_lora: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_grand_central_m4_express: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_itsybitsy_m4_express: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_trinket_m0: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: ev11l78a: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: wio_terminal: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_zero: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_mkrzero: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_nano_33_iot: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

golowanow committed:
boards: riscv: niosv_g: move and convert to HWMv2

Move and convert to HWMv2 riscv/niosv_g board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: riscv: niosv_m: move and convert to HWMv2

Move and convert to HWMv2 riscv/niosv_m board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

fkokosinski committed:
soc/sifive/sifive_freedom: add SiFive Freedom U540 SoC

This commit adds support for the SiFive Freedom U540 SoC for the Zephyr
Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert SiFive HiFive Unleashed to Zephyr HWMv2

This commit converts the SiFive HiFive Unleashed board (`hifive_unleashed`
target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
soc/sifive/sifive_freedom: add SiFive Freedom FU740 SoC

This commit adds support for the SiFive Freedom U740 SoC for the Zephyr
Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert SiFive HiFive Unmatched to Zephyr HWMv2

This commit converts the SiFive HiFive Unmatched board (`hifive_unmatched`
target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

golowanow committed:
soc: arm64: intel_socfpga: Move and convert to HWMv2

Move and convert soc/arm64/intel_socfpga SoC family
`intel_socfpga` configuration to HWMv2 with its SoC
series: `agilex` and `agilex5` and related SoCs:
`intel_socfpga_agilex` and `intel_socfpga_agilex5`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: arm64: intel_socfpga_agilex_socdk: move to HWMv2

Move and convert to HWMv2 `intel_socfpga_agilex_socdk`
board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: arm64: intel_socfpga_agilex5_socdk: move to HWMv2

Move and convert to HWMv2 `intel_socfpga_agilex5_socdk`
board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: arm: intel_socfpga_std: Move and convert to HWMv2

Move and convert to HWMv2 `soc/arm/intel_socfpga_std`
SoC family configuration, its SoC series `cyclonev`,
and SoC `intel_socfpga_std_cyclonev`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: arm: cyclonev_socdk: Move to HWMv2

Move and convert to HWMv2 `cyclonev_socdk` board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
scripts: west: runners: intel_adsp: Adjust path to HWMv2

Adjust script path to HWMv2 location at `soc/intel/intel_adsp/tools`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

fkokosinski committed:
boards: convert SparkFun RED-V Things Plus to Zephyr HWMv2

This commit converts the SparkFun RED-V Things Plus board
(`sparkfun_red_v_things_plus` target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

57300 committed:
cmake: modules: boards: Fix BOARD_ALIAS

`--format` is not a valid argument to `list_boards.py`.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
cmake: modules: boards: Enhance board aliases for HWMv2

To address concerns about lengthy board identifiers in HWMv2, the
proposal is to apply the existing BOARD_ALIAS feature, like so:

   set(<alias>_BOARD_ALIAS <board>/<soc>)

It should then be possible to build with either:

   -DBOARD=<alias>            # expands to <board>/<soc>
   -DBOARD=<alias>/<variant>  # expands to <board>/<soc>/<variant>

However, this wouldn't work out of the box. A board alias can only be
expanded to a board name, without revision or identifier, because the
alias substitution happens after having parsed BOARD as user input -
namely, into BOARD (name), BOARD_REVISION, and BOARD_IDENTIFIER.

Furthermore, this means that in the legacy model, it was possible to
build with `-DBOARD=<alias>@<revision>`, and it would resolve to the
actual board name + revision.

To support both the old and new use cases, we can parse the alias just
like BOARD itself, then concatenate their identifiers as shown above.
Adding a revision works as before, but now it is also possible for the
alias to set its own revision. In this example:

   set(<alias>_BOARD_ALIAS <board>@<rev-A>/<soc>/<variant>)

`<rev-A>` is treated as the default revision, and it can be overridden:

   -DBOARD=<alias>          # expands to <board>@<rev-A>/<soc>/<variant>
   -DBOARD=<alias>@<rev-B>  # expands to <board>@<rev-B>/<soc>/<variant>

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

golowanow committed:
soc: boards: xtensa: intel_adsp_cavs25: Rename with HWMv2

Rename for better HWMv2 use:
 SoC - `intel_tgl_adsp` to `cavs25`

Resulting name changes for boards:
 `intel_adsp_cavs25/intel_tgl_adsp`      -> `intel_adsp_cavs25/cavs25`
 `intel_adsp_cavs25/intel_tgl_adsp/tgph` -> `intel_adsp_cavs25/cavs25/tgph`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: xtensa: intel_adsp_cavs25: Rename board with HWMv2

Rename for better HWMv2 use:
 board - `intel_adsp_cavs25` to `intel_adsp`

Resulting name changes for boards:
 `intel_adsp_cavs25/cavs25`      -> `intel_adsp/cavs25`
 `intel_adsp_cavs25/cavs25/tgph` -> `intel_adsp/cavs25/tgph`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
tests: samples: boards: intel_adsp_cavs25: Rename with HWMv2

Adjust tests and samples to HWMv2 board rename:
     `intel_adsp_cavs25`      -> `intel_adsp/cavs25`
     `intel_adsp_cavs25_tgph` -> `intel_adsp/cavs25/tgph`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
doc: soc: boards: intel_adsp_cavs25: Rename with HWMv2

Adjust documentation to HWMv2 board rename:
  `intel_adsp_cavs25`      -> `intel_adsp/cavs25`
  `intel_adsp_cavs25_tgph` -> `intel_adsp/cavs25/tgph`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

gautierg-st committed:
boards: Convert adi_eval_adin1110ebz to HWM v2

Port the board to HWMv2.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
boards: Convert adi_eval_adin2111ebz to HWM v2

Port the board to HWMv2.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

golowanow committed:
hwmv2: sof: intel_adsp: submanifest provisional link

Provisional link to SOF with adjustment to the HWMv2 new
intel_adsp board names.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

gautierg-st committed:
boards: stm32l4: doc: add zephyr_file to defconfig path

Add :zephyr_file: to the defconfig path for STM32L4 boards

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

manuargue-nxp committed:
boards: nxp: ucans32k1sic: convert to hwmv2

Convert ucans32k1sic board to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

manuargue-nxp committed:
boards: nxp: mr_canhubk3: convert to hwmv2

Convert mr_canhubk3 board to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nordicjm committed:
boards: raspberry_pi: rpi_pico: Use full name for w variant

Updates the name to have the SoC in

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
sample: basic: blinky_pwm: Exclude rpi_pico w variant

Adds an exclusion because this board does not have an LED defined
and will try to use the base board overlay file, resulting in
failure

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nashif committed:
boards: qemu: move qemu_x86 -> x86

Just follow others and drop qemu_ from the name, we have that in the
folder already.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
x86: remove legacy ia32, use atom instead

ia32 is legacy and is just an atom, so deduplicate and use generic
atom soc instead.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
x86: atom: remove soc.h with unused content

None of the code in soc.h is being used anywhere, so remove it and
remove include from various places.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

57300 committed:
cmake: modules: boards: Fix board deprecation for HWMv2

Deprecating boards comes with similar challenges as with board aliases,
since BOARD_DEPRECATED is also set after parsing BOARD as user input.

With this patch, a deprecated board can be properly translated to a
board with identifiers. This opens up the possibility of recording all
legacy board names in `boards/deprecated.cmake`, such as:

   set(mps2_an521_remote_DEPRECATED mps2/an521/cpu1)

Unlike with aliases, though, there are additional restrictions for
building with BOARD=<deprecated>, which only makes this feature suitable
for deprecating v1 boards:

 * BOARD=<deprecated>/<identifier> is never allowed.
 * BOARD=<deprecated>@<revision> is not allowed if the old board already
   corresponds to a new board revision:

   set(<deprecated>_DEPRECATED <new-board>@<new-revision>)

Future enhancements will be needed for deprecating v2 boards.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

nandojve committed:
tests: atmel_sam: Update platform name

This update all atmel_sam entries to use the new platform name related
to sam_e70_xplained and sam_v71_xult.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
samples: atmel_sam: Update platform name

This update all atmel_sam entries to use the new platform name related
to sam_e70_xplained and sam_v71_xult.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

manuargue-nxp committed:
soc: nxp: convert NXP S32 family to hwmv2

Convert NXP S32 family to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nordicjm committed:
maintainers: Update paths for renesas boards/socs

Updates paths to account for changes in hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski committed:
MAINTAINERS: update RISC-V arch area paths

This commit updates the paths assigned to the RISC-V area of maintenance to
include targets based on the SiFive Freedom SoC family.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

gautierg-st committed:
boards: fix documentation for alientek and blues boards

Fix documentation for alientek and blues boards

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

fabiobaltieri committed:
soc: convert microchip_mec to hwmv2

Convert the Microchip MEC soc/ to hardware model v2.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
boards: convert all microchip MEC boards to hwmv2

Convert mec1501modular_assy6885, mec15xxevb_assy6853,
mec172xevb_assy6906 and mec172xmodular_assy6930 to hwmv2.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
boards: adjust few boards/ paths

Adjust few paths in the board document and maintainer file to reflect
the new file location.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

golowanow committed:
MAINTAINERS: intel_adsp: Adjust to HWMv2 move

Adjust intel_adsp paths to HWMv2 move.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: arm64: intel_socfpga: Move boards to subdirectories

Adjust boards directory structure to SoC structure:

 `boards/intel/intel_socfpga_agilex_socdk` ->
     `boards/intel/intel_socfpga/agilex_socdk`
 `boards/intel/intel_socfpga_agilex5_socdk` ->
     `boards/intel/intel_socfpga/agilex5_socdk`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
MAINTAINERS: intel_socfpga: Adjust to HWMv2 move

Adjust intel/intel_socfpga to HWMv2 move.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: arm: intel_socfpga_std: Align board subdirectory

Aligh board directory to other Intel FPGA boards moving
`boards/intel/cyclonev_socdk` -->
`boards/intel/intel_socfpga_std/cyclonev_socdk`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nashif committed:
boards: x86: remove qemu_x86_tiny_768

Having a board to just measure coverage for certain features is a bit of
any overkill. We will instead add overlays to the specific tests this
board mostly covers.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

aescolar committed:
boards: native_sim/posix: Add 64bit versions as variants

For native_sim and native_posix add the 64 bit version
definitions as board variants.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

nashif committed:
soc: x86/atom: imply mmu, do not select it

Board should be able to deselect mmu, so imply instead of strict
selection.

userspace is selected for atom already on the CPU level, so do not
reselect.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nandojve committed:
samples: atmel_sam0: Update platform name

This update all atmel_sam0 related entries to use the new
platform name.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
tests: atmel_sam0: Update platform name

This update all atmel_sam0 related entries to use the new
platform name.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

golowanow committed:
soc: boards: xtensa: intel_adsp_ace: Rename with HWMv2

Rename for better HWMv2 use:

  SoC - `intel_ace15_mtpm` --> `ace15_mtpm`
  SoC - `intel_ace20_lnl`  --> `ace20_lnl`

Resulting shortened name for boards:

  `intel_adsp_ace15_mtpm/intel_ace15_mtpm` -->
  `intel_adsp_ace15_mtpm/ace15_mtpm`

  `intel_adsp_ace20_lnl/intel_ace20_lnl` -->
  `intel_adsp_ace20_lnl/ace20_lnl`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
hwmv2: boards: intel_adsp: Overhaul board configurations

Rename and join configurations for intel_adsp CAVS and ACE
boards to benefit from HWMv2 shortened names with resulting
names change:

  `intel_adsp_ace15_mtpm` --> `intel_adsp/ace15_mtpm`
  `intel_adsp_ace20_lnl`  --> `intel_adsp/ace20_lnl`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
hwmv2: tests: boards: intel_adsp: Adjust board names

Adjust tests to HWMv2 intel_adsp_ace board name changes.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: intel_adsp: tools: pylint compliance workaround

Disable Pylint compliance check warning `R0801:Similar lines`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
doc: boards: intel_adsp: Re-order pages

Index boards/intel/intel_adsp pages.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
samples: bluetooth: hci_uart: Fix wrongly converted board names

Fixes an issue whereby converted board names were wrong

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: neorv32: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: neorv32: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: ambiq: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: apollo4p_evb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: apollo4p_blue_kxr_evb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: ambiq: Add index

Adds a documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update ambiq paths

Updates ambiq SoC and board paths for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: synopsys: Add documentation index

Adds a documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update synopsys entries

Updates entries to account for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: cypress: Add documentation index

Adds documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update paths for hwmv2

Updates paths which have changed in hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: infineon: Add documentation index

Adds documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: microchip_miv: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: m2gl025_miv: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: mpfs_icicle: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: beaglev_fire: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: ti_lm3s6965: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: qemu_cortex_m3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update path for hwmv2

Updates paths for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski committed:
boards: convert SiFive HiFive1 Rev. B to Zephyr HWMv2

This commit converts the SiFive HiFive1 Rev. B board (`hifive1_revb`
target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

FRASTM committed:
soc: soc_legacy: remove the arm/st_stm32 folder

Remove the entire soc/soc_legacy/arm/st_stm32 folder

Signed-off-by: Francois Ramu <francois.ramu@st.com>

nandojve committed:
tests: atmel_sam: adc: Fix sam4e_xpro adc build

This add a missing sam4e_xpro overlay in the tests/drivers/adc for the
sam4e_xpro board.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

erwango committed:
boards: misc: Fixed STM32 based boards doc links

Fix few bad links following board migration.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

danieldegrasse committed:
boards: nxp: add toctree placeholder

Add table of contents tree placeholder for NXP boards, in preparation
for porting boards to HWMv2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

nashif committed:
ci: doc: always report status

Generate status that can be used for enforcing branch checks even if
some jobs get skipped.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: twister status check

Generate one single status for all jobs and also report success when
twister run is being skipped to be able to make this a required check.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

golowanow committed:
soc: board: intel_socfpga_std: Align names to 'Cyclone V'

Align all names to `cyclonev` instead of using `cyclone5`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: arm: intel_socfpga_std: Rename with HWMv2

Rename SoC for better HWMv2 use:
 `intel_socfpga_std_cyclonev` --> `cyclonev`

The resulting shortened board name:
 `cyclonev_socdk/intel_socfpga_std_cyclonev` -->
 `cyclonev_socdk/cyclonev`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

piotrrak committed:
drivers: gpio: renesas_ra: Allow to not specify any interrupt-names.

Fixes compilation when none of interrupts are specified for io_ports.

In practice every RA-series soc will have IRQs, yet it is convinient
still to be able to build code during development of DTS.

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

golowanow committed:
soc: arm64: intel_socfpga: hwmv2: Rename SoCs

Rename SoC for better HWMv2 use:
  `intel_socfpga_agilex`  --> `agilex`
  `intel_socfpga_agilex5` --> `agilex5`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
soc: soc_legacy: mips: Remove out file

Removes a file that is no longer used

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: gaisler: Move Kconfig file

Moves a Kconfig file from hwmv1 to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: silabs_exx32: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32gg_sltb009a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32gg_slwstk6121a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32gg_stk3701a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32hg_slstk3400a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32pg_stk3401a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32wg_stk3800: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32xg24_dk2601b: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32mg_sltb004a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32_thunderboard: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
manifest: Update hal_silabs

Updates the repo to include hwmv2 changes

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32pg_stk3402a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32_radio: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: silab: Add documentation index file

Adds a file with the vendor name for documentation indexing

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Update board names for hwmv2

Updates names used in tests for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski committed:
soc/riscv: convert the QEMU virt RISCV-32 SoC to HWMv2

This commit converts the virt SoC (generic virt machine) to the Zephyr
HWMv2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

erwango committed:
boards: stm32wb: Restore missing .defconfig files

Kconfig.defconfig files has been lost during migration.
Put them back.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

nordicjm committed:
boards: Fix invalid documentation links

Fixes issues with links in documentation

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: Use zephyr_file for file links

Replaces normal text with :zephyr_file: blocks

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski committed:
boards: convert QEMU RISC-V 32 bit board to Zephyr HWMv2

This commit converts the QEMU RISCV-V 32 bit board to Zephyr HWMvW. This
includes the following former targets:
* qemu_riscv32
* qemu_riscv32_smp
* qemu_riscv32_xip

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert QEMU RISC-V 64 bit board to Zephyr HWMv2

This commit converts the QEMU RISCV-V 64 bit board to Zephyr HWMvW. This
includes the following former targets:
* qemu_riscv64
* qemu_riscv64_smp

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert QEMU RV32E board to Zephyr HWMv2

This commit converts the QEMU RV32E board to Zephyr HWMvW. This includes
the following former target: qemu_riscv32e.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

nordicjm committed:
tests: Fix board names

Fixes some invalid board names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand committed:
scripts: ci: introduce soc name check in check_compliance

soc.yml files define SoC names which are used in board.yml.
All SoC names and directories are exported to the build system and
can be referenced using the SoC name as identifier.

Kconfig defines a CONFIG_SOC setting with the same name which can be
used in build system and is selected by the board.

Thus the CONFIG_SOC value can be used to lookup the details of the SoC.

This commit introduces a new compliance check which ensures the SoC name
and the CONFIG_SOC name value are in sync.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: align CONFIG_SOC values to match soc.yml names

This commit aligns CONFIG_SOC values for Nordic nRF SoCs to the names
defined in soc.yml.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

piotrrak committed:
soc: Add Renesas RA6M5 series R7FA6M5BH3CFC

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
dts: renesas ra: Minimal dts for Renesas RA6M5-series R7FA6M5BH3CFC

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
dts: bindings: Draft bindings rv40f flash used in some of socs in RA family

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
boards: Very minimal support for Renesas EK-RA6M5 board.

Supports only samples/basic/minimal.

- pinctrl, gpio partly works, yet minimum of ioports are added,
- pin IRQs for GPIOs aren't set - should work in theory but not tested.

Since this board has two user switches, those will be made available in dts
in follow up change.

Blinky will work with caveat, that seconds are few others magnitude longer;
Reason being proper clocks are not configured yet in this change and
RTC/timers aren't supported yet.

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
soc: renesas ra: RA4M3 series R7FA4M3AF3CFB.

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
dts: renesas ra: RA4M3 series R7FA4M3AF3CFB.

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
board: renesas: EK-RA4M3 preliminary support.

Supports samples/basic/minimal
blinky has same caveats as EK-RA6M5

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
dt-bindings: pinctrl: renesas: Fix value of PSEL_MASK.

PSEL [4:0] is 5 bit bitfield of PmnPFS/PmnPFS_HA/PmnPFS_BY registers which
could have been modeled in C as:

struct PmnPFS_xx {
  union {
     u32 _;
     struct {
        unsigned irrevelant_here: 24;
        unsigned PSEL : 5;
     } fields;
  };
};

Thus mask value should be 0b11111 (0x1f) rather than 0b101, since it is
used as mask:

   ((psel)&PSEL_MASK) << PSEL_POS) /* Note: PSEL_POS = 24 */

in RA_PINCFG macro.

Based of 19.2.5 "Port mn Pin Function Select Register"  RA4M1 Group
Users's: Hardware and other RA Groups manuals.

This change is cherry-pick candidate for main.

Please Note It was _NOT_ tested on `arduino_uno_r4_minima` board

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak commented:
@soburi Hello, could you please take a look in free time especially at https://github.com/zephyrproject-rtos/zephyr/pull/69315/commits/0dbcab24f8f47f57cb87b2be0f01bb669d32cb3b ?
It is most likely bug fix for `arduino_uno_r4_minima` pin values. It's not used in code base, but it is user facing API.

piotrrak committed:
drivers: clock control: renesas ra: Unambigously align code to 32B

Align code to 32B (ICACHE-line) using `.p2align` `.align` statement means different
things for different assemblers and usually not what we really mean.

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

manuargue-nxp committed:
boards: nxp: s32z2xxdc2: convert to hwmv2

Convert `s32z270dc2` boards to hardware model v2. The board has been
renamed to `s32z2xxdc2` to be able to support in the future other
SoCs from this series that can also work on this board.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nordicjm committed:
boards: nordic: nrf5340dk: Fix board names

Updates to new board names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: kernel: tickless: Remove old board name

Removes a duplicated old board name

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

erwango committed:
soc: stm32: Cleanup Kconfig.defconfig files

SOC related symbols are defined in Kconfig.soc

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
soc: stm32: Rename Kconfig SOC_SERIES symbols

SOC symbol is expected to match SOC_SERIES_FOO symbol.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

tejlmand committed:
soc: ti: define SOC name in Kconfig

Add Kconfig SoC name am6234 which matches the SoC name in soc.yml.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: mec: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for mec SoCs defined in soc.yml to
Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: arm: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for arm an547 SoCs defined in soc.yml
to Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: broadcom: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for bcm58402 SoCs defined in soc.yml
to Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

erwango committed:
soc: stm32: Rename series path

Since symbol is used by CMake to locate matching series folder, rename
folders.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

nordicjm committed:
soc: cypress: Add protection guard to file

Adds a protection guard to prevent Kconfigs leaking

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand committed:
boards: nrf51dk/dongle: change SoC to nRF51822

The physical nRF51dk / nRF51dongle hardware contains a nRF51422 SoC.
In Zephyr, only the nRF51822 SoC, is implemented.

In Zephyr, the nRF51422 SoC is build as a nRF51822, therefore use the
nRF51822 SoC model for those kits instead on the non-existing nRF51422.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

nordicjm committed:
soc: xilinx: zync7000: Remove xilinx from soc series name

Removes the vendor name from a Kconfig symbol

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented on boards/renesas/ek_ra4m3/Kconfig.board:
```diff
@@ -0,0 +1,6 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_RA4M3_EK
+	bool "Renesas EK-RA4M3 board"
```
Boards no longer have a choice so
```suggestion
	bool
```

piotrrak replied:
@nordicjm Thanks. Both `Kconfig.board` and `Kconfig.ek_raXX` were present. Simply removed `Kconfig.board` ones.

nordicjm commented on boards/renesas/ek_ra4m3/Kconfig.board:
```diff
@@ -0,0 +1,6 @@
+
+config BOARD_RA4M3_EK
+	bool "Renesas EK-RA4M3 board"
+	depends on SOC_R7FA4M3AF3CFB
```
And need to select the soc e.g.
```suggestion
	select SOC_R7FA4M3AF3CFB
```

nordicjm commented on boards/renesas/ek_ra4m3/Kconfig.board:
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024 Piotr Rak <piotr.rak@gmail.com>
```
Needs to be named `Kconfig.<board>` so `Kconfig.ek_ra4m3`

nordicjm reviewed (commented):
Very quick skim, soc files look fine, changes to board files are needed

golowanow committed:
soc: intel_adsp_cavs: hwmv2: Align SOC_SERIES name and value

Align 'intel_adsp_cavs` SoC Series name and value to match
new HWMv2 compliance check.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: intel_adsp: hwmv2: Align SOC_SERIES_INTEL_ACE name and value

Align `ace` to 'intel_adsp_ace` SoC Series name and value to match
the new HWMv2 compliance check, also renaming:

  SOC_SERIES_INTEL_ACE --> SOC_SERIES_INTEL_ADSP_ACE

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
scripts: west_commands: completion: Add hwmv2 complete to bash

Adds hwmv2 board completetion support to bash

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
scripts: west: list_boards: Fix hwmv2 output

Fixes the output of list_boards to be in csv format rather than
python array format

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
scripts: west_commands: completion: Add hwmv2 complete to zsh

Adds hwmv2 board completetion support to zsh

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
scripts: west_commands: completion: Add hwmv2 complete to fish

Adds hwmv2 board completetion support to fish

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic: nrf9160dk: Fix missing nrf52840 config

Fixes an issue with missing configuration from the nrf52840 on
the nrf9160dk

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

57300 committed:
snippets: Initial HWMv2 support

Very simple adaptation to directly replace the use of HWMv1 board names
with the new identifiers:

   boards:
     <name>/<identifier>:
       append: ...

All snippets in-tree are aligned with this format where needed.

Since the `boards` key supports regex, appends can also be applied to
multiple targets grouped by board name, SoC, CPU cluster, or variant.
Later, this can be followed up with native support for these items in
the snippet schema.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

tejlmand committed:
soc: nxp: fix typo in SoC name

Follow-up: #68778

There is a typo in nxp/kinetics/soc.yml.
The SoC `mk82f215` was added, however the correct SoC is `mk80f25615`,
which this commit corrects.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

piotrrak committed:
dt-bindings: renesas-ra-icu: Add missing RA_ICU_PORT_IRQ13

Required used by some of for RA4/RA6/RA8 soc series.

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

piotrrak committed:
boards: renesas ra: Make ek_ra4m3 manually testable using push buttons

Add port IRQ's, `gpio-keys` for board push buttons and proper aliases in
order to make GPIO IRQs manually testable (eg. using sample/basic/buttons)

Only ioport0 port IRQs are provided for now, as those connected to board
physical buttons.

Signed-off-by: Piotr Rak <piotr.rak@gmail.com>

manuargue-nxp committed:
tests: drivers: gpio: gpio_api_1pin: exclude hifive1

Test seems to be failing consistently on `hifive1` board.
See #69350

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nandojve committed:
soc: atmel: Use new family prefix

The newer HWMv2 impose a different semantic in the family names. This
update from SOC_FAMILY_SAMx to SOC_FAMILY_ATMEL_SAMx to comply with.

Fixes #69046

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nordicjm committed:
boards: qemu: qemu_cortex_a53: Expand names

Expands names to include the SoC

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xen: xenvm: Expand name

Expands name to include identifier

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Expand names

Expands names to include identifiers

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: intel: intel_adsp: Fix issues

Fixes issues with missing protection guards and selections in
wrong files, and one case of missing bools on 2 fields

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand committed:
soc: emsdsp: align SoC name defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC name for emsdp_em4 SoC defined in Kconfig.soc
to to name in soc.yml

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

piotrrak commented:
Rebased on current collab-hwm after its rebase on 3.6.0

soburi commented:
> @soburi Hello, could you please take a look in free time especially at [0dbcab2](https://github.com/zephyrproject-rtos/zephyr/commit/0dbcab24f8f47f57cb87b2be0f01bb669d32cb3b) ? It is most likely bug fix for `arduino_uno_r4_minima` pin values. It's not used in code base, but it is user facing API.

@piotrrak 

Thank you for catching my mistakes.
LGTM.

piotrrak commented:
@soburi  Thank you for looking at this. I've just picked up arduino uno m4 minima and wifi from my mailbox today.  I should be able to test/debug my changes on hardware from now on.

soburi commented:
@piotrrak 

I have created PR #71174.
It still developing, but I think it may help to update your PR.
